app/controllers/api/resource_controller.rb

```diff
@@ -30,19 +30,21 @@ def show
   end
 
   def create
-    payload = params[:data]
-    api_resource = @api_namespace.api_resources.new(properties: payload)
+    api_resource = @api_namespace.api_resources.new(
+      properties: resource_params[:data]
+    )
     if api_resource.save
-      render json: { code: 200, status: 'OK', object: serialize_resource(api_resource) }
-    else
-      render json: { code: 400, status: api_resource.errors.full_messages.to_sentence }
+        render json: { code: 200, status: 'OK', object: serialize_resource(api_resource) }
```

Comment:
@Pralish can we replace the calls to `serialize_resource` with https://github.com/jsonapi-serializer/jsonapi-serializer ?